storage/scratchpad-benchmark/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use diem_config::config::RocksdbConfig;
+use diem_crypto::hash::SPARSE_MERKLE_PLACEHOLDER_HASH;
+use diem_types::{
+    account_address::{AccountAddress, HashAccountAddress},
+    account_state_blob::AccountStateBlob,
+};
+use diemdb::DiemDB;
+use executor_types::ProofReader;
+use rand::Rng;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use storage_interface::DbReader;
+
+type SparseMerkleTree = scratchpad::SparseMerkleTree<AccountStateBlob>;
+
+fn gen_random_blob<R: Rng>(size: usize, rng: &mut R) -> AccountStateBlob {
+    let mut v = vec![0u8; size];
+    rng.fill(v.as_mut_slice());
+    AccountStateBlob::from(v)
+}
+
+pub fn run_benchmark(num_updates: usize, blob_size: usize, db_dir: PathBuf) {
+    let db = DiemDB::open(
+        &db_dir,
+        false, /* readonly */
+        None,  /* pruner */
+        RocksdbConfig::default(),
+    )
+    .expect("DB should open.");
+
+    let mut rng = ::rand::thread_rng();
+
+    let updates = (0..num_updates)
+        .into_iter()
+        .map(|_| {
+            (
+                AccountAddress::random(),
+                gen_random_blob(blob_size, &mut rng),
+            )
+        })
+        .collect::<Vec<_>>();
+
+    let version = db.get_latest_version().unwrap();
+    let account_state_proofs = updates
```

Comment:
Most will be non-existent proofs? 

I think an executor-benchmark run with a lot of accounts created will be a better simulation of the real life situation.

Reply:
I can create with a ratio of existent proofs. 
Executor-benchmark uses storage-service which is hard to measure the perf of smt only. Also, it cannot create a super large db in a short time.

Reply:
The ProofReader is actually in-mem reads all the time even in the executor-benchmark, so it should be good?

Reply:
It is indeed. But for executor benchmark if you want to create a 1M account db you need to wait for 1h?

Reply:
executor-benchmark is a superset with some execution that we don't want to measure. I don't see why it is better for this purpose besides it has more existent proofs.

Reply:
But 1M accounts is the scenario where we really want to know the improvements. It'll be nice if we can persist the DB with the 1M accounts and share it across runs.